### DISCUSSION TITLE
"Guest requested more data than was available" Error

### DISCUSSION THREAD

**Discussion by jeljelih** [risc0/risc0#1283] [Q&A] [answered]
When I pass 3 u64 as an input to Guest, I obtain the following error: 
```
thread 'main' panicked at risc0/zkvm/src/host/server/exec/syscall.rs:377:9:
assertion left == right failed: Guest requested more data than was available
  left: 0
 right: 4
```

**Host code**
```
    let (x_A, y_A) = (139743900, 35631200);
    let d = 70;

    let input = (x_A, y_A, d);
    let env = ExecutorEnv::builder().write(&input).unwrap().build().unwrap();
```

**Guest code**
```
pub fn main() {
    let (x_A, y_A, d): (u64, u64, u64) = env::read();
}
```

Any idea how to solve this issue?

**Comment by flaub** [comment, marked as answer]
I don't think you can assume that `let d = 70;` will be a `u64` on the host side. It's likely the compiler will select `usize` in this case which is going to be 64-bits on most host machines, but it will be 32-bits in the guest.
